test/mlkem_internal_test.c

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2024 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#include <openssl/opensslconf.h>
+#ifndef OPENSSL_NO_STDIO
+# include <stdio.h>
+#endif
+
+#ifndef OPENSSL_NO_MLKEM
+# include <crypto/mlkem.h>
+#endif
+
+#include <string.h>
+#include "testutil.h"
+#include "testutil/output.h"
+
+int main(void)
+{
+#ifndef OPENSSL_NO_MLKEM
+    uint8_t out_encoded_public_key[OSSL_MLKEM768_PUBLIC_KEY_BYTES];
+    uint8_t out_ciphertext[OSSL_MLKEM768_CIPHERTEXT_BYTES];
+    uint8_t out_shared_secret[OSSL_MLKEM768_SHARED_SECRET_BYTES];
+    uint8_t out_shared_secret2[OSSL_MLKEM768_SHARED_SECRET_BYTES];
+    ossl_mlkem768_private_key private_key;
+    ossl_mlkem768_public_key public_key;
+    ossl_mlkem768_public_key recreated_public_key;
+    uint8_t *p1, *p2;
+
+    /* enable TEST_* API */
+    test_open_streams();
+
+    /* first, generate a key pair */
+    if (!ossl_mlkem768_generate_key(out_encoded_public_key, NULL, &private_key))
+        return 1;
+    /* public key component to be created from private key */
+    ossl_mlkem768_public_from_private(&public_key, &private_key);
+    /* try to re-create public key structure from encoded public key */
+    ossl_mlkem768_recreate_public_key(out_encoded_public_key, &recreated_public_key);
```

Comment:
This function can fail so we should check the return code

Reply:
Thanks for the catch. So changed in https://github.com/openssl/openssl/pull/25848/commits/7b7cdfb693855255e65ff6968c491f55688b0996